lib/crypto/sig_verifier_test.go

```diff
@@ -0,0 +1,222 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package crypto_test
+
+import (
+	"testing"
+
+	"github.com/ChainSafe/gossamer/internal/log"
+	"github.com/ChainSafe/gossamer/lib/crypto"
+	"github.com/ChainSafe/gossamer/lib/crypto/ed25519"
+	"github.com/ChainSafe/gossamer/lib/crypto/secp256k1"
+	"github.com/ChainSafe/gossamer/lib/crypto/sr25519"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestSigVerifierEd25519(t *testing.T) {
+	t.Helper()
+	signs := make([]*crypto.Signing, 2)
+
+	for i := 0; i < 2; i++ {
+		msg := []byte("Hello")
+		key, err := ed25519.GenerateKeypair()
+		require.NoError(t, err)
+
+		sign, err := key.Sign(msg)
+		require.NoError(t, err)
+
+		signs[i] = &crypto.Signing{
+			PubKey:     key.Public().Encode(),
+			Sign:       sign,
+			Msg:        msg,
+			VerifyFunc: ed25519.VerifySignature,
+		}
+	}
+	signVerify := crypto.NewSignatureVerifier(log.New())
```

Comment:
You'll have to import `io` as well at the top

```suggestion
	signVerify := crypto.NewSignatureVerifier(log.New(log.SetWriter(io.Discard)))
```

This applies to other `log.New()` in this file.